modules/tinymce/src/core/main/ts/textpatterns/core/BlockPattern.ts

```diff
@@ -57,10 +57,18 @@ const applyPattern = (editor: Editor, match: BlockPatternMatch): boolean => {
 
 // Finds a matching pattern to the specified text
 const findPattern = <P extends Pattern>(patterns: P[], text: string): Optional<P> => {
+  const sortedPatterns = sortPatterns(patterns);
   const nuText = text.replace(Unicode.nbsp, ' ');
-  return Arr.find(patterns, (pattern) => text.indexOf(pattern.start) === 0 || nuText.indexOf(pattern.start) === 0);
+  return Arr.find(sortedPatterns, (pattern) => text.indexOf(pattern.start) === 0 || nuText.indexOf(pattern.start) === 0);
 };
 
+const sortPatterns = <T extends Pattern>(patterns: T[]): T[] => Arr.sort(patterns, (a, b) => {
```

Comment:
Functions shouldn't be used before the are defined if we can help it... so can you please move this somewhere that makes sense above `findPattern`?

Reply:
Also for consistency, the generic should be called `P` here as it is above.